Ginger/GingerCore/Drivers/SeleniumDriver.cs

```diff
@@ -3687,17 +3685,35 @@ private ObservableList<ElementInfo> GetAllElementsFromPage(string path, List<eEl
                         }
 
                         //filter element if needed
-                        eElementType foundElementType = GetElementTypeEnum(el);
+
+                 
+                        eElementType elementTypeEnum = GetElementTypeEnum(el);
                         if (filteredElementType != null && filteredElementType.Count > 0)
                         {
-                            if (!filteredElementType.Contains(foundElementType))
+                            if (!filteredElementType.Contains(elementTypeEnum))
                             {
                                 continue;
                             }
                         }
 
-                        ElementInfo foundElemntInfo = null;
-                        foundElemntInfo = GetElementInfoWithIWebElement(el, htmlNode, path, learnFullElementInfoDetails);
+                        HTMLElementInfo foundElemntInfo = new HTMLElementInfo();
```

Comment:
foundElemntInfo creation needs to be inside  if (learnPartialElementInfoDetails) because on else it will create new elementInfo object in old function